src/KafkaFlow.BatchConsume/BatchConsumeMiddleware.cs

```diff
@@ -5,19 +5,28 @@
     using System.Linq;
     using System.Threading;
     using System.Threading.Tasks;
-
-    internal class BatchConsumeMiddleware : IMessageMiddleware, IDisposable
+    using KafkaFlow.Configuration;
+    using KafkaFlow.Consumers;
+    using KafkaFlow.Observer;
+
+    internal class BatchConsumeMiddleware
+        : IMessageMiddleware,
+            ISubjectObserver<WorkerStoppedSubject, VoidObject>,
+            IDisposable
     {
         private readonly SemaphoreSlim dispatchSemaphore = new(1, 1);
 
         private readonly int batchSize;
         private readonly TimeSpan batchTimeout;
         private readonly ILogHandler logHandler;
+        private readonly IConsumerConfiguration consumerConfiguration;
 
         private readonly List<IMessageContext> batch;
         private CancellationTokenSource dispatchTokenSource;
+        private Task<Task> dispatchTask;
 
         public BatchConsumeMiddleware(
+            IConsumerMiddlewareContext workerContext,
```

Comment:
This isn't only related to the worker context, right?

Reply:
renamed. :eyes: